flutter/lib/src/web_session_updater.dart

```diff
@@ -0,0 +1,58 @@
+import '../sentry_flutter.dart';
+import 'native/sentry_native_binding.dart';
+
+class WebSessionUpdater implements BeforeSendEventObserver {
+  final SentryFlutterOptions _options;
+  final SentryNativeBinding _nativeBinding;
+
+  WebSessionUpdater(this._nativeBinding, this._options);
+
+  @override
+  void onBeforeSendEvent(SentryEvent event, Hint hint) {
+    _updateSessionFromEvent(event);
+  }
+
+  void _updateSessionFromEvent(SentryEvent event) async {
+    if (_options.enableAutoSessionTracking == false) {
+      _options.logger(
+        SentryLevel.info,
+        'Auto session tracking for web disabled.',
+      );
+      return;
+    }
+
+    bool crashed = event.level == SentryLevel.fatal;
+    bool errored = false;
+
+    if (event.exceptions?.isNotEmpty == true) {
+      errored = true;
+
+      for (final exception in event.exceptions!) {
+        if (exception.mechanism?.handled == false) {
+          crashed = true;
+          break;
+        }
+      }
+    }
+
+    // A session is updated and that session update is sent in only one of the two following scenarios:
+    // 1. Session with non terminal status and 0 errors + an error occurred -> Will set error count to 1 and send update
+    // 2. Session with non terminal status and 1 error + a crash occurred -> Will set status crashed and send update
+    final session = await _nativeBinding.getSession();
+    if (session == null) {
+      return;
+    }
+
+    final sessionNonTerminal = session['status'] == 'ok';
+    final shouldUpdateAndSend =
+        (sessionNonTerminal && session['errors'] == 0) ||
```

Comment:
In the case:

- The current session is `ok` and it has no errors.
- The event is neither crashed, nor errored.

We would update in this case and set the session to the same values, at least how I understood it.

Is this needed or can we just ignore session updated if the event is not crashed nor errored?

Reply:
> Is this needed or can we just ignore session updated if the event is not crashed nor errored?

good point, we can just ignore session handling if the event doesn't have an error, this is also in JS but I overlooked it